scripts/generate_skeleton/generate_skeleton.py

```diff
@@ -0,0 +1,440 @@
+    if not name.isidentifier():
+        raise ValueError(f"Name '{name}' is not a valid Python identifier")
+
+    # Check for Python keywords
+    if keyword.iskeyword(name) or keyword.issoftkeyword(name):
+        raise ValueError(f"Name '{name}' is a Python keyword and cannot be used")
+
+    # Enforce snake_case (no uppercase letters)
+    if name != name.lower():
+        raise ValueError("Name must be lowercase (snake_case)")
+
+    # Additional check for valid characters (letters, numbers, underscores only)
+    if not name.replace("_", "").isalnum():
+        raise ValueError("Name can only contain letters, numbers, and underscores")
+
+
+def validate_category(category: str) -> None:
+    """Validate category name for security.
+
+    Args:
+        category: The category to validate
+
+    Raises:
+        ValueError: If category is invalid
+    """
+    if not category:
+        raise ValueError("Category cannot be empty")
+
+    # Check for directory traversal attempts
+    if "/" in category or "\\" in category:
+        raise ValueError("Category cannot contain path separators (/, \\)")
+
+    if "." in category:
+        raise ValueError("Category cannot contain dots (.)")
+
+    # Enforce snake_case (no uppercase letters, allow underscores)
+    if category != category.lower():
+        raise ValueError("Category must be lowercase (snake_case)")
+
+    # Allow letters, numbers, and underscores
+    if not category.replace("_", "").isalnum() or not category[0].isalpha():
+        raise ValueError("Category can only contain letters, numbers, and underscores")
+
+
+def get_existing_categories(skeleton_type: str) -> list[str]:
+    """Get list of existing categories for the given skeleton type.
+
+    Args:
+        skeleton_type: Either 'component' or 'pipeline'
+
+    Returns:
+        List of existing category directory names
+    """
+    base_dir = Path(f"{skeleton_type}s")
+    if not base_dir.exists():
+        return []
+
+    return [item.name for item in base_dir.iterdir() if item.is_dir() and not item.name.startswith((".", "_"))]
+
+
+def generate_core_files(skeleton_type: str, category: str, name: str):
+    """Generate core files for skeleton based on type.
+
+    Args:
+        skeleton_type: Type of skeleton ('component' or 'pipeline')
+        category: Category name
+        name: Skeleton name
+
+    Returns:
+        Dict of filename to content mappings
+    """
+    env = _get_template_env()
+    current_date = datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    # Prepare type-specific template variables
+    if skeleton_type == "component":
+        decorator_name = "component"
+        decorator_params = (
+            '    base_image="python:3.11",\n'
+            '    # packages_to_install=["numpy", "pandas"],  # Add your dependencies here'
+        )
+        param_default = ""
+        param_extras = ",\n    # Add your output artifacts here\n" "    # output_artifact: dsl.Output[dsl.Artifact]"
```

Comment:
This line is weird. Is the implicit concatenation intentional? How about using triple quotes to increase readability?